libs/frontend/feature-command/src/lib/component/replace-page-root.spec.ts

```diff
@@ -0,0 +1,196 @@
+
+  const compareComponent = (
+    compA: ISerializedComponent | undefined,
+    compB: ISerializedComponent | undefined,
+    ignoredProps?: Set<keyof ISerializedComponent>,
+  ) => {
+    // Make sure compA and compB are either both truthy or both falsy.
+    expect(!!compA).toEqual(!!compB)
+
+    if (compA) {
+      Object.keys(compA).forEach((key) => {
+        const prop = key as keyof ISerializedComponent
+        if (ignoredProps?.has(prop)) {
+          return
+        } else if (key === 'children') {
+          // Make sure the children of compA and compB are either both truthy or both falsy.
+          expect(!!compA.children).toEqual(!!compB?.children)
+
+          // Recursively compare children.
+          compA.children?.forEach((childA, index) => {
+            const childB = compB?.children?.[index]
+            compareComponent(childA, childB, ignoredProps)
+          })
+        } else {
+          expect(compA[prop]).toEqual(compB?.[prop])
+        }
+      })
+    }
+  }
+
+  beforeEach(() => {
+    siteString = JSON.stringify(mockSerializedSite)
+    site = deserializeSite(siteString) as ISite
+    page = site.pages['/home']
+
+    // Add the following component to the site:
+    // <HorizontalContainer name="copyMe">
+    //   <H1>Hello</H1>
+    //   <Text>World</Text>
+    //   <VerticalContainer>
+    //     <Input />
+    //   </VerticalContainer>
+    // </HorizontalContainer>
+    const copiedComponentName = 'copyMe'
+    const newComponent = mockSerializedComponent(site, {
+      name: copiedComponentName,
+      tag: Tag.Div,
+      parentId: page.root.id,
+      children: [
+        mockSerializedComponent(site, {
+          name: 'H1',
+          tag: Tag.H1,
+          content: 'Hello',
+        }),
+        mockSerializedComponent(site, {
+          name: 'Text',
+          tag: Tag.Div,
+          content: 'World',
+        }),
+        mockSerializedComponent(site, {
+          name: 'VerticalContainer',
+          tag: Tag.Div,
+          children: [
+            mockSerializedComponent(site, {
+              name: 'Input',
+              tag: Tag.Input,
+            }),
+          ],
+        }),
+      ],
+    })
+    applyAddComponent(site, mockAddComponentData(newComponent))
+
+    // Assign copied component
+    const copiedComp = page.root.children?.find(
+      (child) => child.name === copiedComponentName,
+    )
+    if (!copiedComp) {
+      throw new Error('Cannot find the component to be copied')
+    }
+    copiedComponent = copiedComp
+
+    // Prepare replace page root data
+    replacePageRootData = {
+      pageRoute: page.route,
+      oldRoot: makeRemoveComponentData(site, page.root, true),
+      replacementComponent: {
+        name: copiedComponent.name,
+        tag: copiedComponent.tag,
+        content: copiedComponent.content,
+        sourceId: copiedComponent.id,
+        parentId: '',
+      },
+    }
+  })
+
+  it('should replace page root with the specified component', () => {
```

Comment:
It could be useful to add a test where a new page's root is pasted to, but this sufficient for now.